lib/multipart.js

```diff
@@ -51,11 +51,16 @@ Multipart.prototype.setHeaders = function (chunked) {
   }
 
   var header = self.request.getHeader('content-type')
-  var contentType = (!header || header.indexOf('multipart') === -1)
-    ? 'multipart/related'
-    : header.split(';')[0]
 
-  self.request.setHeader('content-type', contentType + '; boundary=' + self.boundary)
+  if (!header || header.indexOf('multipart') === -1) {
+    self.request.setHeader('content-type', 'multipart/related; boundary=' + self.boundary)
+  } else {
+    if (header.indexOf('boundary') !== -1) {
+      self.boundary = header.replace(/.*boundary=([^\s;])+.*/, '$1')
```

Comment:
Thanks for pointing that out @assaf the bug was fixed here https://github.com/request/request/pull/1505 I had a faulty test for that case.